AzureCalling/External/Calling/CallingContext.swift

```diff
@@ -63,537 +30,34 @@ class CallingContext: NSObject {
-            completionHandler(.success(()))
-        }
-    }
-
-    func pauseLocalVideoStream(completionHandler: @escaping (Result<Void, Error>) -> Void) {
-        if isCameraPreferredOn {
-            self.stopLocalVideoStream { [weak self] result in
-                guard let self = self else {
-                    return
-                }
-                if case .failure(let error) = result {
-                    print("ERROR: Local video failed to pause. \(error)")
-                    completionHandler(.failure(error))
-                    return
-                }
-                self.isVideoOnHold = true
-                print("Local video paused successfully")
-                completionHandler(.success(()))
-            }
-        }
-    }
-
-    func switchCamera(completionHandler: @escaping (Result<Void, Error>) -> Void) {
-        guard let localVideoStream = localVideoStream else {
-            return
-        }
-
-        let currentCamera = localVideoStream.source
-        let flip: CameraFacing = currentCamera.cameraFacing == .front ? .back : .front
-
-        if let flipCamera = availableCameras[flip] {
-            localVideoStream.switchSource(camera: flipCamera) { error in
-                if error != nil {
-                    print("ERROR: It was not possible to switch camera. \(error!)")
-                    completionHandler(.failure(error!))
-                    return
-                }
-                print("Camera switch successful")
-                completionHandler(.success(()))
-            }
-        } else {
-            print("ERROR: The other camera is absent.")
-            completionHandler(.failure(NSError()))
-        }
-    }
-
-    func mute(completionHandler: @escaping (Result<Void, Error>) -> Void) {
-        self.call?.mute(completionHandler: { (error) in
-            if error != nil {
-                print("ERROR: It was not possible to mute. \(error!)")
-                completionHandler(.failure(error!))
-                return
-            }
-            print("Mute successful")
-            completionHandler(.success(()))
-        })
-    }
-
-    func unmute(completionHandler: @escaping (Result<Void, Error>) -> Void) {
-        self.call?.unmute(completionHandler: { (error) in
-            if error != nil {
-                print("ERROR: It was not possible to unmute. \(error!)")
-                completionHandler(.failure(error!))
-                return
-            }
-            print("Unmute successful")
-            completionHandler(.success(()))
-        })
-    }
-
-    private func setupCalling(completionHandler: @escaping (Result<Void, Error>) -> Void) {
-        callClient = CallClient()
-        self.setupDeviceManager(completionHandler: completionHandler)
-    }
-
-    private func setupCallAgent(displayName: String, completionHandler: @escaping (Result<Void, Error>) -> Void) {
+    func getTokenCredential() -> CommunicationTokenCredential? {
```

Comment:
@AmyL219 I believe when we were chatting with CallingSDK team they recommended us not pass in `nil` as token. Should we fix this for the hero sample?

Reply:
Good point. What CallingSDK team said is we should fetch token first. I can schedule a meeeting regarding this

Reply:
Lets not block the PR cause the rest of the changes are good!